src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -234,7 +241,11 @@ TokenizeSnippets::TokenizeSnippets() {
     continuation_strategy strategy = continuation_strategy::reset;
     auto label = std::make_shared<pattern::op::Label>(pattern::any_input(),
         [](const std::shared_ptr<const Node> &n) {
-            return GetSnippetsNodeType(n) != SnippetsNodeType::SkippedByPlugin && AppropriateForSubgraph(n);
+            // todo: This is a temprorary work-around. remove when custom MHA tokenization pass is implemented
+            return (GetSnippetsNodeType(n) != SnippetsNodeType::SkippedByPlugin ||
+                    ov::is_type<opset1::MatMul>(n) || ov::is_type<opset1::Transpose>(n))
```

Comment:
Why do we need to explicitly check specific ops here. Even if this is temporary WA broader explanation is needed.

Reply:
Added a more detailed description.
The problem here is that the markup pass always skips matmuls and transposes, but I'm not sure that we need to modify it until matmul&transpose tokenization si not supported through general pipeline.